management/src/main/java/org/ehcache/management/registry/DefaultCollectorService.java

```diff
@@ -84,7 +85,9 @@ public synchronized void start(ServiceProvider serviceProvider) {
     cacheManager = serviceProvider.getService(CacheManagerProviderService.class).getCacheManager();
     scheduledExecutorService = serviceProvider.getService(ExecutionService.class).getScheduledExecutor(configuration.getCollectorExecutorAlias());
 
-    cacheManager.registerListener(this);
+    if (cacheManager instanceof ObservableCacheManager) {
```

Comment:
@cljohnso @lorban @chrisdennis @ljacomet : I have an alternative solution:

When I created the `CacheManagerProviderService`, the goal was to be able within the services to access a cache manager. Firstly because if the m&m needs (to listen on some events).

So perhaps instead of checking whether the CM is an instance of Observable, and silently discard a feature the user explicitly installed, why don't we change the `CacheManagerProviderService ` so that we have something like this:

```
CacheManager cacheManager = serviceProvider.getService(CacheManagerProviderService.class).getCacheManager();

ObservableCacheManager cacheManager = serviceProvider.getService(CacheManagerProviderService.class).getObservableCacheManager();
```
`getObservableCacheManager` would do a sanity check and throw an exception if the CM is not observable.

or...

`ObservableCacheManager cacheManager = serviceProvider.getService(CacheManagerProviderService.class).getCacheManager(ObservaleCacheManager.class);`

and the method `getCacheManager(Class<T extends CacheManager>)` would "unwrap" to the wanted class or throw.

The idea behind this is that a service gets what it needs, and if he can't then it crashes simply because it won't work anyway. I think it's not the responsibility of the service to check / know all the different potential flavors of a CM (because at the end there could be more than just the Observable flavor). The service just need to ask what it needs to work properly, and the init just need to fail if what is needed cannot be provided. It would be the responsibility of the developer to fix such error.

Management services are optional (theoretically because jsr107 depends on it at the moment, at least until #747 is fixed). So if somebody register such services within a CM that is not observable, it should fail at init (so when the cache manager provider would throw an exception), because the required feature won't work.

Let me know what you decided: I am interested in following the changes in the management module :-)